### PR TITLE
Add focus-visible style for NcCheckboxRadioSwitch

### PR DESCRIPTION
**Pull request by Fenn-CS**
Improve accesibility by making focus indicator visible.

Resolves : https://github.com/nextcloud/photos/issues/1671

### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
7f1a8522565d214125291c78487d02e1 | Screenshot from 2023-06-09 16-59-51

### 🏁 Checklist

- [ ] ⛑️ Tests are included or are not applicable
- [ ] 📘 Component documentation has been extended, updated or is not applicable

### CODE REVIEW

File: src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue [hunk 1]
```diff
@@ -270,17 +270,16 @@ export default {
...
 			:disabled="disabled"
 			:indeterminate="indeterminate"
 			:name="name"
 			:type="inputType"
 			:value="value"
 			class="checkbox-radio-switch__input"
 			@change="onToggle">
-		<label :for="id" class="checkbox-radio-switch__label">
-			<div class="checkbox-radio-switch__icon">
+		<button class="checkbox-radio-switch__button" @click="onToggle">
```
Comment (Fenn-CS) [inline src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:281]: `<label>` cannot receive any of the events in the class of `focus*` hence the only way to make this focus-able it use an element that can give us that and here a button seems appropriate.

Buttons, get some focus already however unpredictable but that would be addressed in another issue here : https://github.com/nextcloud/nextcloud-vue/issues/2824

Comment (skjnldsv) [reply src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:281]: Sorry, I understand the reasoning, but label+input semantics are much more important than a focus-visible :)

Using keyboard, the focus-visible will be on th einput.
Which allows you to style the label by using a css direct sibling selector: `input:focus-visible + label`

**Review by skjnldsv — CHANGES_REQUESTED** [review]

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Awesome @Fenn-CS  :)

File: src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue [hunk 2]
```diff
@@ -583,6 +583,10 @@ export default {
 		height: var(--icon-size);
 	}
 
+	&__input:focus-visible + label {
+		outline: 2px solid var(--color-main-text) !important;
```
Comment (raimund-schluessler) [inline src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:587]: Is `outline` the standard for showing `focus-visible`? It creates issues when multiple elements are close together:
checkbox

Would it be an option to use `border` instead? When we give the `label` a `transparent` border by default, reduce the padding by `2px` in all directions and switch to a colored border when `focus-visible`, it creates a similar effect, but without the overlapping:

checkbox_border

```css
.checkbox-radio-switch__input:focus-visible + label[data-v-1b902817] {
  /* outline: 2px solid var(--color-main-text) !important; */
  border: 2px solid var(--color-main-text) !important;
}

.checkbox-radio-switch__label[data-v-1b902817] {
  /* padding: 4px 14px; */
  padding: 2px 12px;
  border: 2px solid transparent;
}
```

Comment (Fenn-CS) [reply src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:587]: @szaimen I guess you are suggesting we start using `--color-primary-element` in this, right?

Comment (szaimen) [reply src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue:587]: Yes 👍

Comment (Fenn-CS) [reply src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue]: @szaimen done!

**Review by Pytal — APPROVED** [review]

**Pull request merged**